beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -48,17 +48,29 @@ func (vs *Server) setExecutionData(ctx context.Context, blk interfaces.BeaconBlo
 	if err != nil {
 		log.WithError(err).Warn("Proposer: failed to check if builder can be used")
 	} else if canUseBuilder {
-		h, err := vs.getPayloadHeaderFromBuilder(ctx, slot, idx)
+		builderPayload, err := vs.getPayloadHeaderFromBuilder(ctx, slot, idx)
 		if err != nil {
 			builderGetPayloadMissCount.Inc()
 			log.WithError(err).Warn("Proposer: failed to get payload header from builder")
 		} else {
-			blk.SetBlinded(true)
-			if err := blk.Body().SetExecution(h); err != nil {
-				log.WithError(err).Warn("Proposer: failed to set execution payload")
-			} else {
-				return nil
+			localPayload, err := vs.getExecutionPayload(ctx, slot, idx, blk.ParentRoot(), headState)
```

Comment:
what happens if this is still bellatrix. There is no block value, then this call will fail.

Reply:
yeah, I still need to add coverage here. Thanks for pointing this out!